src/email/outbound.js

```diff
@@ -0,0 +1,303 @@
+const { createConnection } = require('net');
+const { resolveMx } = require('dns');
+
+const { DKIMSign } = require('dkim-signer');
+const MailComposer = require("nodemailer/lib/mail-composer");
+const ParseEmailAddress = require("email-addresses");
+const { Logging } = require('../log');
+
+const CRLF = '\r\n';
+
+const log = Logging.get("outbound");
+
+module.exports = function (options) {
+    options = options || {};
+
+    const dkimPrivateKey = (options.dkim || {}).privateKey;
+    const dkimKeySelector = (options.dkim || {}).keySelector || 'dkim';
+    const devPort = options.devPort || -1;
+    const devHost = options.devHost || 'localhost';
+    const smtpPort = options.smtpPort || 25;
+    const smtpHost = options.smtpHost || -1;
+
+
+    function groupRecipients(recipients) {
+        let groups = {};
+        let host;
+        const recipientsLength = recipients.length;
+        for (let i = 0; i < recipientsLength; i++) {
+            host = ParseEmailAddress.parseOneAddress(recipients[i]).domain;
+            (groups[host] || (groups[host] = [])).push(recipients[i]);
+        }
+        return groups;
+    }
+
+    /**
+     * connect to domain by MX record
+     */
+    function connectMx(domain, callback) {
+        if (devPort === -1) { // not in development mode -> search the MX
```

Comment:
A small request of mine to make this more readable is to make the larger if block a top level block.
Rather than

```
if (condition) {
... lots
... and
... lots
... of
... code
} else {
  ... less code
}
```

you can do:
```
if (!condition) {
   ...less code
   return;
}
... larger block of code
```

The advantage here is that it saves you an else, and it's more readable for me :)